PWGHF/TableProducer/treeCreatorD0ToKPi.cxx

```diff
@@ -390,11 +405,36 @@ struct HfTreeCreatorD0ToKPi {
         double yD = yD0(candidate);
         double eD = eD0(candidate);
         double ctD = ctD0(candidate);
+        int candType = 0;
         if (candidate.isSelD0()) {
-          fillTable(candidate, prong0, prong1, 0, invMassD0ToPiK(candidate), cosThetaStarD0(candidate), ctD, yD, eD, candidate.flagMcMatchRec(), candidate.originMcRec());
+          if (candidate.flagMcMatchRec() == (1 << DecayType::D0ToPiK)) {
+            candType |= o2::aod::full::IsSignal;
+          } else {
+            if (candidate.flagMcMatchRec() == -(1 << DecayType::D0ToPiK)) {
+              candType |= o2::aod::full::ReflectedD0;
+            }
+          }
+          if (candidate.originMcRec() == RecoDecay::OriginType::Prompt) {
+            candType |= o2::aod::full::Prompt;
+          } else {
+            candType |= o2::aod::full::NonPrompt;
+          }
```

Comment:
> Thanks a lot for the comment @fgrosa . I removed the CandidateType column in the HFCANDD0FULL table and replaced the CandidateSelFlag in the HFCANDD0LITE as you pointed.

Hi @Mingyu3360715 thanks a lot! If I am not mistaken, you should also remove the `FlagMc` and `OriginMcRec` ones, since they are also not needed anymore. Finally, you could move this part of the code inside the `fillTable` method in case of `fillCandidateLiteTable` to 1) avoid code duplication and 2) avoid the computation of `candType` in case of full tables.